lib/src/rules/prefer_relative_imports.dart

```diff
@@ -93,34 +75,19 @@ class _Visitor extends SimpleAstVisitor<void> {
       return false;
     }
 
-    // Is the package: import referencing the current package?
-    var segments = uri.pathSegments;
-    if (segments.isEmpty) return false;
-
-    if (parsedPubspec == null) {
-      String content;
-      try {
-        content = pubspecFile.readAsStringSync();
-        // ignore: avoid_catches_without_on_clauses
-      } catch (_) {}
-      parsedPubspec = _parseYaml(content);
-    }
-
-    return parsedPubspec['name'] == segments[0];
+    final source = node.uriElement?.library?.source;
+    if (source == null) return false;
+    // todo (pq): should context.package.contains(source)work?
+    return path.isWithin(context.package.root, source.fullName);
```

Comment:
Hmm, I don't follow the logic here. How does this trigger when there's a package self-reference in lib/, but not in bin/? I don't know what `context.package.root` holds.

If you think this logic is correct (which it very well could be), perhaps add more positive and negative tests to make sure it works in several different types of situations?

Reply:
The lib check is handled by the `isInLibFolder` bool we set in `visitCompilationUnit` and use to short-circuit this visit which is to say, if we're not in `lib/` we shouldn't even get here.

That said, some more tests are a great idea. 👍